[LogicApps] New Api version

visharm opened the pull request:
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

visharm committed:
Added 2019-05-01 stable version

visharm committed:
Updated API version in the json files

visharm committed:
Added 2019 version in readme

AutorestCI commented:
# Automation for azure-sdk-for-python

AutorestCI commented:
# Automation for azure-sdk-for-java

AutorestCI commented:
# Automation for azure-sdk-for-go

azuresdkci commented:
Can one of the admins verify this patch?

praries880 commented:
@visharm Kindly fix the following:
* Model validation errors: https://dev.azure.com/azure-sdk/public/_build/results?buildId=66071&view=logs&j=8d2f9c49-cd83-5a9d-f3fe-2fd651afa742&t=43324f1f-6724-5d61-785f-f0ac9fd256a1&l=125****

* Remove the unreferenced files: https://dev.azure.com/azure-sdk/public/_build/results?buildId=66071&view=logs&j=f10f5372-40ed-54fc-caa4-b1cb5bd1252b&t=d3b2b841-a93b-5fa5-47c3-d8c4f6eae6e3&l=15

* Refer the correct json file : https://dev.azure.com/azure-sdk/public/_build/results?buildId=66071&view=logs&j=688669d0-441c-57c3-cf6d-f89a22ccfa5d&t=923b5a90-e462-5994-c166-86509c6bf5f2&l=129

And there are a bunch of other CI erros as well.

visharm commented:
Thanks Pareekshit.

I will fix the errors (2 and 3).

But why am I getting model errors when I have just reused the old swagger file for the new version. Haven’t yet added new APIs yet.
Was the old swagger file not correct?

Also is there way I can validate the build locally before pushing the change to remote branch?

Thanks,
Vinita

From: Pareekshit Singh <notifications@github.com>
Sent: Friday, July 12, 2019 1:17 PM
To: Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com>
Cc: Vinita Sharma <Vinita.Sharma@microsoft.com>; Mention <mention@noreply.github.com>
Subject: Re: [Azure/azure-rest-api-specs] [LogicApps] New Api version (#6601)


@visharm<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fvisharm&data=02%7C01%7CVinita.Sharma%40microsoft.com%7C91880116d86c40a0bef608d70705dac1%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636985594039360160&sdata=ROs68YF2hrX4qmWO9PymoARqKooEQaAUQC3DRzbrKrE%3D&reserved=0> Kindly fix the following:

  *   Model validation errors: https://dev.azure.com/azure-sdk/public/_build/results?buildId=66071&view=logs&j=8d2f9c49-cd83-5a9d-f3fe-2fd651afa742&t=43324f1f-6724-5d61-785f-f0ac9fd256a1&l=125<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdev.azure.com%2Fazure-sdk%2Fpublic%2F_build%2Fresults%3FbuildId%3D66071%26view%3Dlogs%26j%3D8d2f9c49-cd83-5a9d-f3fe-2fd651afa742%26t%3D43324f1f-6724-5d61-785f-f0ac9fd256a1%26l%3D125&data=02%7C01%7CVinita.Sharma%40microsoft.com%7C91880116d86c40a0bef608d70705dac1%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636985594039370156&sdata=Bz0x6O4TaWEw%2F2ADlbxA8k1CXexPbLgAvo%2Fh4UZskpE%3D&reserved=0>****
  *   Remove the unreferenced files: https://dev.azure.com/azure-sdk/public/_build/results?buildId=66071&view=logs&j=f10f5372-40ed-54fc-caa4-b1cb5bd1252b&t=d3b2b841-a93b-5fa5-47c3-d8c4f6eae6e3&l=15<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdev.azure.com%2Fazure-sdk%2Fpublic%2F_build%2Fresults%3FbuildId%3D66071%26view%3Dlogs%26j%3Df10f5372-40ed-54fc-caa4-b1cb5bd1252b%26t%3Dd3b2b841-a93b-5fa5-47c3-d8c4f6eae6e3%26l%3D15&data=02%7C01%7CVinita.Sharma%40microsoft.com%7C91880116d86c40a0bef608d70705dac1%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636985594039370156&sdata=xLacieWydODEkUiWO3wYBoNoC%2FJo%2FsVG7wtaWXa30b0%3D&reserved=0>
  *   Refer the correct json file : https://dev.azure.com/azure-sdk/public/_build/results?buildId=66071&view=logs&j=688669d0-441c-57c3-cf6d-f89a22ccfa5d&t=923b5a90-e462-5994-c166-86509c6bf5f2&l=129<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdev.azure.com%2Fazure-sdk%2Fpublic%2F_build%2Fresults%3FbuildId%3D66071%26view%3Dlogs%26j%3D688669d0-441c-57c3-cf6d-f89a22ccfa5d%26t%3D923b5a90-e462-5994-c166-86509c6bf5f2%26l%3D129&data=02%7C01%7CVinita.Sharma%40microsoft.com%7C91880116d86c40a0bef608d70705dac1%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636985594039380152&sdata=bJiK8CQyQrNtRDlMI3avX2C627F%2Bk1orWgsdUmdHMrk%3D&reserved=0>

And there are a bunch of other CI erros as well.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F6601%3Femail_source%3Dnotifications%26email_token%3DAI3GSZUKP42W5CHI65FK7FDP7DRCTA5CNFSM4IBWNGP2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODZ2YUEY%23issuecomment-511019539&data=02%7C01%7CVinita.Sharma%40microsoft.com%7C91880116d86c40a0bef608d70705dac1%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636985594039380152&sdata=facoXlikGmjDbeDyZJ6CNYZjFD7U4P4TOJqX1071yaE%3D&reserved=0>, or mute the thread<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAI3GSZSA3ITLS6D3PCV62FLP7DRCTANCNFSM4IBWNGPQ&data=02%7C01%7CVinita.Sharma%40microsoft.com%7C91880116d86c40a0bef608d70705dac1%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636985594039390145&sdata=qF43DVqQ%2BVeI1ynqMKKGcSeTSfsE5as5LkeLsx4xueE%3D&reserved=0>.

visharm committed:
Fixed errors

ryansbenson commented on specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:
```diff
@@ -0,0 +1,10048 @@
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscription
```
This was removed from the previous api version and should be added back:

"default": {
            "description": "Logic error response describing why the operation failed.",
            "schema": {
              "$ref": "#/definitions/ErrorResponse"
            }
          }

ryansbenson replied:
This was removed from the previous api version and should be added back:

"default": {
"description": "Logic error response describing why the operation failed.",
"schema": {
"$ref": "#/definitions/ErrorResponse"
}
}

ryansbenson commented on specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:
```diff
@@ -0,0 +1,10048 @@
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscription
```
This was removed from the previous api version and should be added back:

"default": {
"description": "Logic error response describing why the operation failed.",
"schema": {
"$ref": "#/definitions/ErrorResponse"
}
}

visharm replied:
I just compared the swagger diff between 2019-05-01 to 2018-07-01. The only change is the version name. 
The new swagger contains all this, where is this removed?

ryansbenson reviewed (commented):
Just some minor things that didn't come over from the last api version.

visharm commented:
I don’t see its removed.

The new api version has all this change.

From: ryansbenson <notifications@github.com>
Sent: Friday, July 12, 2019 4:16 PM
To: Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com>
Cc: Vinita Sharma <Vinita.Sharma@microsoft.com>; Mention <mention@noreply.github.com>
Subject: Re: [Azure/azure-rest-api-specs] [LogicApps] New Api version (#6601)


@ryansbenson commented on this pull request.

Just some minor things that didn't come over from the last api version.

________________________________

In specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F6601%23discussion_r303174529&data=02%7C01%7CVinita.Sharma%40microsoft.com%7C4988a3f8d2c847ba453108d7071edc12%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636985701433386582&sdata=IEaT8tvbfHbqE9UK%2BYx1nZmPV49UIJmv9ah5yQV2gwk%3D&reserved=0>:

> +    },

+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Logic/workflows/{workflowName}/runs/{runName}/actions/{actionName}/requestHistories": {

+      "get": {

+        "tags": [

+          "WorkflowRunActions"

+        ],

+        "operationId": "WorkflowRunActionRequestHistories_List",

+        "description": "List a workflow run request history.",

+        "x-ms-examples": {

+          "List a request history": {

+            "$ref": "./examples/WorkflowRunActionRequestHistories_List.json"

+          }

+        },

+        "parameters": [

+          {

+            "$ref": "#/parameters/subscription

This was removed from the previous api version and should be added back:

"default": {
"description": "Logic error response describing why the operation failed.",
"schema": {
"$ref": "#/definitions/ErrorResponse"
}
}

________________________________

In specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F6601%23discussion_r303174582&data=02%7C01%7CVinita.Sharma%40microsoft.com%7C4988a3f8d2c847ba453108d7071edc12%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636985701433396578&sdata=6uJvz%2FqvtCQVAQRCfNVFF%2FOojhPArwLbqQI0eNLI90w%3D&reserved=0>:

> +    },

+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Logic/workflows/{workflowName}/runs/{runName}/actions/{actionName}/requestHistories": {

+      "get": {

+        "tags": [

+          "WorkflowRunActions"

+        ],

+        "operationId": "WorkflowRunActionRequestHistories_List",

+        "description": "List a workflow run request history.",

+        "x-ms-examples": {

+          "List a request history": {

+            "$ref": "./examples/WorkflowRunActionRequestHistories_List.json"

+          }

+        },

+        "parameters": [

+          {

+            "$ref": "#/parameters/subscription

This was removed from the previous api version and should be added back:

"default": {
"description": "Logic error response describing why the operation failed.",
"schema": {
"$ref": "#/definitions/ErrorResponse"
}
}

________________________________

In specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F6601%23discussion_r303174692&data=02%7C01%7CVinita.Sharma%40microsoft.com%7C4988a3f8d2c847ba453108d7071edc12%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636985701433406575&sdata=gke8Rr958njNigkImaFC5yBcRQI1%2FCWCWQyUV0mGoBs%3D&reserved=0>:

> +    },

+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Logic/workflows/{workflowName}/runs/{runName}/actions/{actionName}/requestHistories": {

+      "get": {

+        "tags": [

+          "WorkflowRunActions"

+        ],

+        "operationId": "WorkflowRunActionRequestHistories_List",

+        "description": "List a workflow run request history.",

+        "x-ms-examples": {

+          "List a request history": {

+            "$ref": "./examples/WorkflowRunActionRequestHistories_List.json"

+          }

+        },

+        "parameters": [

+          {

+            "$ref": "#/parameters/subscription

This was removed from the previous api version and should be added back:

"default": {
"description": "Logic error response describing why the operation failed.",
"schema": {
"$ref": "#/definitions/ErrorResponse"
}
}

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F6601%3Femail_source%3Dnotifications%26email_token%3DAI3GSZUGJ42Y7VSGR236YOLP7EGB3A5CNFSM4IBWNGP2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB6LEPBA%23pullrequestreview-261506948&data=02%7C01%7CVinita.Sharma%40microsoft.com%7C4988a3f8d2c847ba453108d7071edc12%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636985701433406575&sdata=wxA%2B6nZj7yjxawM8SubgwqzEsVGNRHcASr%2BqjF4qmb4%3D&reserved=0>, or mute the thread<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAI3GSZRJW3ICJABUS6XEEN3P7EGB3ANCNFSM4IBWNGPQ&data=02%7C01%7CVinita.Sharma%40microsoft.com%7C4988a3f8d2c847ba453108d7071edc12%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636985701433416571&sdata=bTGFxMvJBFdsUGFU4VavCsTnRhlXs2SPzVBXAgaOSpo%3D&reserved=0>.

visharm committed:
Added ISE APIs

majastrz commented:
>         "responses": {

You need to add a "default" response to cover all errors in your Swagger. This is missing from new and existing paths. Take a look at blueprint swagger, we have implemented that there.

---
Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5110 in 4ee2e9a. [](commit_id = 4ee2e9a00005b08cf9dda630a3f9b3b22db9b107, deletion_comment = False)

majastrz commented:
>           }

Your comments suggest that this is an asynchronous/long-running operation, but you are missing x-ms-long-running-operation extension. Please add it to all long running operations.

---
Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5280 in 4ee2e9a. [](commit_id = 4ee2e9a00005b08cf9dda630a3f9b3b22db9b107, deletion_comment = False)

majastrz commented:
> 	    "patch": {

Will patch always be synchronous?

---
Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5283 in 4ee2e9a. [](commit_id = 4ee2e9a00005b08cf9dda630a3f9b3b22db9b107, deletion_comment = False)

majastrz commented:
>         "responses": {

In addition to the default response I mentioned earlier, you should also return 204 when the object doesn't exist. (DELETEs should never return 404.)

---
Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5367 in 4ee2e9a. [](commit_id = 4ee2e9a00005b08cf9dda630a3f9b3b22db9b107, deletion_comment = False)

majastrz commented:
>   	"/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Logic/integrationServiceEnvironments/{integrationServiceEnvironmentName}/health/network": {

The ARM RPC requires paths to follow the format type1/name1/type2/name2. `/health/network` violates that rule.

I think the best option would be to rename it to `networkHealthStatuses` or something similar and implement a route to GET a single networkHealthStatuses object. This is the preferred option because it automatically works with RBAC (User with reader role will be able to read status without any issues.). This would also avoid the usage of `additionalProperties` in the response definition and would allow you to leverage paging when the number of status items gets too big.


---
Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5516 in 4ee2e9a. [](commit_id = 4ee2e9a00005b08cf9dda630a3f9b3b22db9b107, deletion_comment = False)

majastrz commented:
>         "responses": {

Can you add a fake 200 response that models what the final GET response on the status of the long running operation looks like. (This is a convention we use to work around limitations in swagger 2.0.)

---
Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5762 in 4ee2e9a. [](commit_id = 4ee2e9a00005b08cf9dda630a3f9b3b22db9b107, deletion_comment = False)

majastrz commented:
>     "/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Logic/integrationServiceEnvironments/{integrationServiceEnvironmentName}/managedApis/{apiName}/apiOperations" : {

Where's the path to GET a single apiOperation?

---
Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5829 in 4ee2e9a. [](commit_id = 4ee2e9a00005b08cf9dda630a3f9b3b22db9b107, deletion_comment = False)

majastrz commented:
>               "$ref": "#/definitions/IntegrationServiceEnvironmentSkuDefinition"

The SKU contract documented at [https://armwiki.azurewebsites.net/api_contracts/SKUUpdateContract%E2%80%8B%E2%80%8B.html?q=sku](https://armwiki.azurewebsites.net/api_contracts/SKUUpdateContract%E2%80%8B%E2%80%8B.html?q=sku) states that this should return a list of SKUs, but it seems to be returning a single one.

---
Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5412 in 4ee2e9a. [](commit_id = 4ee2e9a00005b08cf9dda630a3f9b3b22db9b107, deletion_comment = False)

majastrz commented:
>         "name" : {

This overloads the meaning of the top-level `name` property on the resource. Other RPs typically call something like this by prefixing with resource type name. Can you rename to `apiName` or something similar?

---
Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:7389 in 4ee2e9a. [](commit_id = 4ee2e9a00005b08cf9dda630a3f9b3b22db9b107, deletion_comment = False)

majastrz commented:
> 	    "delete": {

Will delete need to be async?

---
Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5334 in 4ee2e9a. [](commit_id = 4ee2e9a00005b08cf9dda630a3f9b3b22db9b107, deletion_comment = False)

majastrz reviewed (changes requested):
I added a number of comments. Please take a look.

visharm commented:
> "/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Logic/integrationServiceEnvironments/{integrationServiceEnvironmentName}/health/network": {
> 
> 
> The ARM RPC requires paths to follow the format type1/name1/type2/name2. /health/network violates that rule.
> I think the best option would be to rename it to networkHealthStatuses or something similar and implement a route to GET a single networkHealthStatuses object. This is the preferred option because it automatically works with RBAC (User with reader role will be able to read status without any issues.). This would also avoid the usage of additionalProperties in the response definition and would allow you to leverage paging when the number of status items gets too big.
> 
> Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5516 in 4ee2e9a. [](commit_id = 4ee2e9a, deletion_comment = False)



> ":

visharm commented:
> "patch": {
> 
> 
> Will patch always be synchronous?
> 
> Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5283 in 4ee2e9a. [](commit_id = 4ee2e9a, deletion_comment = False)

yes

visharm commented:
> "/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Logic/integrationServiceEnvironments/{integrationServiceEnvironmentName}/health/network": {
> 
> 
> The ARM RPC requires paths to follow the format type1/name1/type2/name2. /health/network violates that rule.
> I think the best option would be to rename it to networkHealthStatuses or something similar and implement a route to GET a single networkHealthStatuses object. This is the preferred option because it automatically works with RBAC (User with reader role will be able to read status without any issues.). This would also avoid the usage of additionalProperties in the response definition and would allow you to leverage paging when the number of status items gets too big.
> 
> Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5516 in 4ee2e9a. [](commit_id = 4ee2e9a, deletion_comment = False)

We were planning to implement multiple APIs to get health of various resources. Since this API is to get the health of the network so its called network/health. Also since we have already GA these APIs it not possible to change it now. But will sure keep in mind for the future APIs.

visharm commented:
> "delete": {
> 
> 
> Will delete need to be async?
> 
> Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5334 in 4ee2e9a. [](commit_id = 4ee2e9a, deletion_comment = False)

Delete is async in the background (means we create a job to deprovision it in the background) but for user the resource gets deleted immediately.

visharm commented:
> "delete": {
> Will delete need to be async?
> Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5334 in 4ee2e9a. [](commit_id = 4ee2e9a, deletion_comment = False)
> 
> Delete is async in the background (means we create a job to deprovision it in the background) but for user the resource gets deleted immediately.

so to answer your question delete is NOT async.

visharm commented:
> "name" : {
> 
> 
> This overloads the meaning of the top-level name property on the resource. Other RPs typically call something like this by prefixing with resource type name. Can you rename to apiName or something similar?
> 
> Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:7389 in 4ee2e9a. [](commit_id = 4ee2e9a, deletion_comment = False)

Since these APIs are all GA I cannot change these now.

visharm commented:
> "/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Logic/integrationServiceEnvironments/{integrationServiceEnvironmentName}/managedApis/{apiName}/apiOperations" : {
> 
> 
> Where's the path to GET a single apiOperation?
> 
> Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5829 in 4ee2e9a. [](commit_id = 4ee2e9a, deletion_comment = False)

I don't think so we have an API to get single API operation. I have not implemented this API but I think there is no need to get a single API operation. Is this a blocker?

visharm commented:
> "$ref": "#/definitions/IntegrationServiceEnvironmentSkuDefinition"
> 
> 
> The SKU contract documented at https://armwiki.azurewebsites.net/api_contracts/SKUUpdateContract%E2%80%8B%E2%80%8B.html?q=sku states that this should return a list of SKUs (matching the documented contract), but it seems to be returning a single one.
> 
> Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5412 in 93887d8. [](commit_id = 93887d8, deletion_comment = False)

yes I fixed that earlier.

visharm commented:
> "responses": {
> 
> 
> You need to add a "default" response to cover all errors in your Swagger. This is missing from new and existing paths. Take a look at blueprint swagger, we have implemented that there.
> 
> Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5110 in 4ee2e9a. [](commit_id = 4ee2e9a, deletion_comment = False)

Could you please point to an example, I don't see one in the blueprint
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/blueprint/resource-manager/Microsoft.Blueprint/preview/2018-11-01-preview/blueprintAssignment.json

visharm commented:
> "responses": {
> 
> 
> Can you add a fake 200 response that models what the final GET response on the status of the long running operation looks like. (This is a convention we use to work around limitations in swagger 2.0.)
> 
> Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5762 in 4ee2e9a. [](commit_id = 4ee2e9a, deletion_comment = False)

I am not sure why do I need to add a 200 fake response for this when this API returns a async operation header to track the long running operation and user get do a get on the long running operation to track its progress.

visharm commented:
> "responses": {
> 
> 
> You need to add a "default" response to cover all errors in your Swagger. This is missing from new and existing paths. Take a look at blueprint swagger, we have implemented that there.
> 
> Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5110 in 4ee2e9a. [](commit_id = 4ee2e9a, deletion_comment = False)

I have added default response to the new APIs but if its missing from the existing ones then why didn't the validation fail before or someone pointed it out earlier.
Do you want me to add it in all existing APIs as well?

visharm committed:
Added examples and addressed CR comments

ravbhatnagar commented:
> > "responses": {
> > Can you add a fake 200 response that models what the final GET response on the status of the long running operation looks like. (This is a convention we use to work around limitations in swagger 2.0.)
> > Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5762 in [4ee2e9a](https://github.com/Azure/azure-rest-api-specs/commit/4ee2e9a00005b08cf9dda630a3f9b3b22db9b107). [](commit_id = [4ee2e9a](https://github.com/Azure/azure-rest-api-specs/commit/4ee2e9a00005b08cf9dda630a3f9b3b22db9b107), deletion_comment = False)
> 
> I am not sure why do I need to add a 200 fake response for this when this API returns a async operation header to track the long running operation and user get do a get on the long running operation to track its progress.

Yes, please do add it. It is required.

ravbhatnagar commented:
> > "responses": {
> > You need to add a "default" response to cover all errors in your Swagger. This is missing from new and existing paths. Take a look at blueprint swagger, we have implemented that there.
> > Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5110 in [4ee2e9a](https://github.com/Azure/azure-rest-api-specs/commit/4ee2e9a00005b08cf9dda630a3f9b3b22db9b107). [](commit_id = [4ee2e9a](https://github.com/Azure/azure-rest-api-specs/commit/4ee2e9a00005b08cf9dda630a3f9b3b22db9b107), deletion_comment = False)
> 
> I have added default response to the new APIs but if its missing from the existing ones then why didn't the validation fail before or someone pointed it out earlier.
> Do you want me to add it in all existing APIs as well?

yes, please add it. Your service would already be returning the errors in the ARM compliant format. So adding them now will not cause any harm. we are just going to make the SDK better.

ravbhatnagar commented:
> > "/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Logic/integrationServiceEnvironments/{integrationServiceEnvironmentName}/managedApis/{apiName}/apiOperations" : {
> > Where's the path to GET a single apiOperation?
> > Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5829 in [4ee2e9a](https://github.com/Azure/azure-rest-api-specs/commit/4ee2e9a00005b08cf9dda630a3f9b3b22db9b107). [](commit_id = [4ee2e9a](https://github.com/Azure/azure-rest-api-specs/commit/4ee2e9a00005b08cf9dda630a3f9b3b22db9b107), deletion_comment = False)
> 
> I don't think so we have an API to get single API operation. I have not implemented this API but I think there is no need to get a single API operation.
> Just checked with the lead, only Logic App designer uses apiOperations, it is not a tracked resource. So there is no need to add single apiOperation

Doesnt matter if its a tracked resource or not. Point GET must be supported as per the ARM RPC.

ravbhatnagar commented:
> > "/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Logic/integrationServiceEnvironments/{integrationServiceEnvironmentName}/health/network": {
> > The ARM RPC requires paths to follow the format type1/name1/type2/name2. /health/network violates that rule.
> > I think the best option would be to rename it to networkHealthStatuses or something similar and implement a route to GET a single networkHealthStatuses object. This is the preferred option because it automatically works with RBAC (User with reader role will be able to read status without any issues.). This would also avoid the usage of additionalProperties in the response definition and would allow you to leverage paging when the number of status items gets too big.
> > Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5516 in [4ee2e9a](https://github.com/Azure/azure-rest-api-specs/commit/4ee2e9a00005b08cf9dda630a3f9b3b22db9b107). [](commit_id = [4ee2e9a](https://github.com/Azure/azure-rest-api-specs/commit/4ee2e9a00005b08cf9dda630a3f9b3b22db9b107), deletion_comment = False)
> 
> We were planning to implement multiple APIs to get health of various resources. Since this API is to get the health of the network so its called network/health.
> Also the dictionary it returns is just for the health of the network.
> Also since we have already GA these APIs it not possible to change it now. But will sure keep in mind for the future APIs.

This will not work in ARM. RBAC will be broken among other things. ARM expects the following segments - after provider namespace, its resourceType/{instance} segments. In this case, network will become the instance of health resource type. You do realize that RBAC is broken for this GA API, and there is no criteria for not fixing a broken API, if its GA.

ravbhatnagar commented on specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:
```diff
@@ -5114,459 +5125,2029 @@
+        }
+      }
+	  },
+	  "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Logic/integrationServiceEnvironments/{integrationServiceEnvironmentName}/skus": {
```
SKUs is a provider level API. It should be /subscriptions/{subscription-id}/providers/Microsoft.Logic/skus. And in this response, the skus for all resource types under the provider must be returned.

ravbhatnagar replied:
Please let me know if you need access to the ARM sku contract.

visharm replied:
As discussed, this API needs to return all Integration Service Environment (resource) Skus for Microsoft Logic RP. So it is modelled at resource level.
The monitoring team want this API for auto-scale of ISE and they wanted it this way so I guess other RPs are also doing something similar.

It won't be possible to change it now without breaking monitoring on ISE so I can remove it from swagger if that's your concern and change it later in consent with Monitoring and ARM team.

visharm replied:
As discussed waiting for the response you get from Andy's team on this.

ravbhatnagar commented on specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:
```diff
@@ -5114,459 +5125,2029 @@
+        }
+      }
+    },     
+  	"/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Logic/integrationServiceEnvironments/{integrationServiceEnvironmentName}/health/network": {
```
Incorrect segments as noted in previous reviews. RBAC will not work. Please fix.

visharm replied:
The API confirms with REST standards. Today we just have one health component which is network so we are returning network health here.
As far as RBAC is concerned, anybody who has GET permissions on Integration Service Environment should be able to get the network health. In this case RBAC does not need to be on the health but on the ISE.

ravbhatnagar replied:
Customer can have their own custom RBAC. but anyway, since network is an instance of health resource type, as per my other comment, it can be modeled as a parameter which is an enum where the allowed value is "network". In future if you want to support more, you can add to the enum.

visharm replied:
As discussed closing this.

ravbhatnagar commented on specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:
```diff
@@ -5114,459 +5125,2029 @@
+        }
+      }
+    },
+    "/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Logic/integrationServiceEnvironments/{integrationServiceEnvironmentName}/managedApis/{apiName}/operationStatuses/{operationId}" : {
```
API to track LRO need not be modeled in swagger, this is picked from header.

visharm replied:
Okay removed.

ravbhatnagar commented on specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:
```diff
@@ -5114,459 +5125,2029 @@
+          }
+        }
+      },
+	    "delete": {
```
Does this delete all other resources which reference the ISE?

visharm replied:
Yes all the resources created in the ISE are deleted.

ravbhatnagar replied:
What if the user does not have permissions to delete those resources? Is there a way from ISE to know all the resources which are created in ISE? I dont see it in the object model.

visharm replied:
It is as simple as when the container is getting deleted, all the artifacts within that container gets deleted. Now the user who is deleting this container knows that all LA etc created in this container will get deleted. Its the same principle followed everywhere else also.

ravbhatnagar commented on specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:
```diff
@@ -5114,459 +5125,2029 @@
+      "type": "object",
+      "properties": {}
+    },
+    "ResourceReference": {
```
this should just be the id, other two properties are not needed. they are made read-only, which is fine. but ideally these are not needed

visharm replied:
This is existing definition, I would leave it as it as.

ravbhatnagar replied:
ok

ravbhatnagar commented on specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:
```diff
@@ -5603,805 +7169,1033 @@
-            "$ref": "#/definitions/RetryHistory"
-          },
-          "description": "Gets the retry histories."
+        "integrationServiceEnvironment" :{
```
why is this needed? ManagedAPI is already under an ISE.

visharm replied:
Its because the same object model is already used in shared environment.

ravbhatnagar replied:
not sure I follow. If I am doing a GET on /integratedServiceEnvironments/ise1/managedApis/api1. Then why should the response of  api1 include the payload of ise1?

visharm replied:
Is there any contract which dictates that RP cannot return this? The object just contains a reference of the ISE it is present in.

visharm replied:
As discussed closing this.

ravbhatnagar reviewed (commented):
@visharm - few comments, please take a look

visharm commented:
> "/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Logic/integrationServiceEnvironments/{integrationServiceEnvironmentName}/health/network": {
> The ARM RPC requires paths to follow the format type1/name1/type2/name2. /health/network violates that rule.
> I think the best option would be to rename it to networkHealthStatuses or something similar and implement a route to GET a single networkHealthStatuses object. This is the preferred option because it automatically works with RBAC (User with reader role will be able to read status without any issues.). This would also avoid the usage of additionalProperties in the response definition and would allow you to leverage paging when the number of status items gets too big.
> Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5516 in 4ee2e9a. [](commit_id = 4ee2e9a, deletion_comment = False)
> 
> We were planning to implement multiple APIs to get health of various resources. Since this API is to get the health of the network so its called network/health.
> Also the dictionary it returns is just for the health of the network.
> Also since we have already GA these APIs it not possible to change it now. But will sure keep in mind for the future APIs.
> 
> This will not work in ARM. RBAC will be broken among other things. ARM expects the following segments - after provider namespace, its resourceType/{instance} segments. In this case, network will become the instance of health resource type. You do realize that RBAC is broken for this GA API, and there is no criteria for not fixing a broken API, if its GA.

The API confirms with REST standards. Today we just have one health component which is network so we are returning network health here.
As far as RBAC is concerned, anybody who has GET permissions on Integration Service Environment should be able to get the network health. In this case RBAC does not need to be on the health but on the ISE.

visharm commented:
> "responses": {
> You need to add a "default" response to cover all errors in your Swagger. This is missing from new and existing paths. Take a look at blueprint swagger, we have implemented that there.
> Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5110 in 4ee2e9a. [](commit_id = 4ee2e9a, deletion_comment = False)
> 
> I have added default response to the new APIs but if its missing from the existing ones then why didn't the validation fail before or someone pointed it out earlier.
> Do you want me to add it in all existing APIs as well?
> 
> yes, please add it. Your service would already be returning the errors in the ARM compliant format. So adding them now will not cause any harm. we are just going to make the SDK better.

Sure added

nschonni commented on specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:
```diff
@@ -0,0 +1,11851 @@
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscription
```
decription -> description
```suggestion
            "description" : "The sku name"
```

nschonni commented on specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:
```diff
@@ -0,0 +1,11851 @@
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscription
```
occured -> occurred
```suggestion
          "description": "The error if any occurred during the operation."
```

visharm replied:
changed

nschonni commented on specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:
```diff
@@ -0,0 +1,11851 @@
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscription
```
occured -> occurred
```suggestion
          "description": "The error if any occurred during the operation."
```

visharm replied:
changed

visharm commented:
> "responses": {
> 
> 
> Can you add a fake 200 response that models what the final GET response on the status of the long running operation looks like. (This is a convention we use to work around limitations in swagger 2.0.)
> 
> Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5762 in 4ee2e9a. [](commit_id = 4ee2e9a, deletion_comment = False)

why do I need to return a fake response of 200 when the API does not return that.
I added the swagger for tracking the async long running operation. Also I am confused Gaurav has a comment to remove that (long running operation from swagger)
Is there any standard which dictates this?

visharm committed:
Cleaned up code and addressed comments

visharm committed:
Fixed model errors

visharm committed:
fixed model errors

visharm committed:
fixed typo

visharm committed:
fixed more errors

ravbhatnagar commented:
> > "responses": {
> > Can you add a fake 200 response that models what the final GET response on the status of the long running operation looks like. (This is a convention we use to work around limitations in swagger 2.0.)
> > Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5762 in [4ee2e9a](https://github.com/Azure/azure-rest-api-specs/commit/4ee2e9a00005b08cf9dda630a3f9b3b22db9b107). [](commit_id = [4ee2e9a](https://github.com/Azure/azure-rest-api-specs/commit/4ee2e9a00005b08cf9dda630a3f9b3b22db9b107), deletion_comment = False)
> 
> why do I need to return a fake response of 200 when the API does not return that.
> I added the swagger for tracking the async long running operation. Also I am confused Gaurav has a comment to remove that (long running operation from swagger)
> Is there any standard which dictates this?

It is for clients to know the shape of the final response of the particular API. That is why it is required. Majority of RPs, if not all, out of 200+ RPs have added this.

ravbhatnagar commented:
> > "/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Logic/integrationServiceEnvironments/{integrationServiceEnvironmentName}/health/network": {
> > The ARM RPC requires paths to follow the format type1/name1/type2/name2. /health/network violates that rule.
> > I think the best option would be to rename it to networkHealthStatuses or something similar and implement a route to GET a single networkHealthStatuses object. This is the preferred option because it automatically works with RBAC (User with reader role will be able to read status without any issues.). This would also avoid the usage of additionalProperties in the response definition and would allow you to leverage paging when the number of status items gets too big.
> > Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5516 in [4ee2e9a](https://github.com/Azure/azure-rest-api-specs/commit/4ee2e9a00005b08cf9dda630a3f9b3b22db9b107). [](commit_id = [4ee2e9a](https://github.com/Azure/azure-rest-api-specs/commit/4ee2e9a00005b08cf9dda630a3f9b3b22db9b107), deletion_comment = False)
> > We were planning to implement multiple APIs to get health of various resources. Since this API is to get the health of the network so its called network/health.
> > Also the dictionary it returns is just for the health of the network.
> > Also since we have already GA these APIs it not possible to change it now. But will sure keep in mind for the future APIs.
> > This will not work in ARM. RBAC will be broken among other things. ARM expects the following segments - after provider namespace, its resourceType/{instance} segments. In this case, network will become the instance of health resource type. You do realize that RBAC is broken for this GA API, and there is no criteria for not fixing a broken API, if its GA.
> 
> The API confirms with REST standards. Today we just have one health component which is network so we are returning network health here.
> As far as RBAC is concerned, anybody who has GET permissions on Integration Service Environment should be able to get the network health. In this case RBAC does not need to be on the health but on the ISE.

Can you make the last segment of this API (which currently is network) to be a parameter like {name}. This can be an enum where the only allowed value is network.

visharm commented:
> "/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Logic/integrationServiceEnvironments/{integrationServiceEnvironmentName}/health/network": {
> The ARM RPC requires paths to follow the format type1/name1/type2/name2. /health/network violates that rule.
> I think the best option would be to rename it to networkHealthStatuses or something similar and implement a route to GET a single networkHealthStatuses object. This is the preferred option because it automatically works with RBAC (User with reader role will be able to read status without any issues.). This would also avoid the usage of additionalProperties in the response definition and would allow you to leverage paging when the number of status items gets too big.
> Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5516 in 4ee2e9a. [](commit_id = 4ee2e9a, deletion_comment = False)
> We were planning to implement multiple APIs to get health of various resources. Since this API is to get the health of the network so its called network/health.
> Also the dictionary it returns is just for the health of the network.
> Also since we have already GA these APIs it not possible to change it now. But will sure keep in mind for the future APIs.
> This will not work in ARM. RBAC will be broken among other things. ARM expects the following segments - after provider namespace, its resourceType/{instance} segments. In this case, network will become the instance of health resource type. You do realize that RBAC is broken for this GA API, and there is no criteria for not fixing a broken API, if its GA.
> 
> The API confirms with REST standards. Today we just have one health component which is network so we are returning network health here.
> As far as RBAC is concerned, anybody who has GET permissions on Integration Service Environment should be able to get the network health. In this case RBAC does not need to be on the health but on the ISE.
> 
> Can you make the last segment of this API (which currently is network) to be a parameter like {name}. This can be an enum where the only allowed value is network.

I can but my only concern is the response object model is specific to network.

visharm commented:
> "responses": {
> Can you add a fake 200 response that models what the final GET response on the status of the long running operation looks like. (This is a convention we use to work around limitations in swagger 2.0.)
> Refers to: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:5762 in 4ee2e9a. [](commit_id = 4ee2e9a, deletion_comment = False)
> 
> why do I need to return a fake response of 200 when the API does not return that.
> I added the swagger for tracking the async long running operation. Also I am confused Gaurav has a comment to remove that (long running operation from swagger)
> Is there any standard which dictates this?
> 
> It is for clients to know the shape of the final response of the particular API. That is why it is required. Majority of RPs, if not all, out of 200+ RPs have added this.

Added

yungezz commented on specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:
```diff
@@ -0,0 +1,12241 @@
+          "WorkflowRunActions"
+        ],
+        "operationId": "WorkflowRunActionRepetitions_ListExpressionTraces",
+        "description": "Lists a workflow run expression t
```
are you sure modelAsString is false?

yungezz replied:
why not reuse DayOfWeek enum?

yungezz replied:
set modelAsString false will make adding new enum value risky, might cause sdk breaking change.

yungezz replied:
wrong format

yungezz commented:
HI @ravbhatnagar @majastrz  is ARM in change requesting status or signing off status?

yungezz reviewed (changes requested):
waiting for arm conclusion

ravbhatnagar commented:
Major issues -
1. Support for point GET on managedApiOperations - Violates ARM RPC, not consistent with other Azure services.  since this is metadata kind of information and no use case outside of portal UX scenario (Logic Apps designer). Still this should have been addresses for consistency sake as it gets hard for customers to know when they can do a POINT gET in azure and when it wont work. Its trivial enough to enable this for uber platform consistency. Engineering Manager (Rohithaa) on Logic apps is fine with this behavior and does not want to fix this.
2. SKU API- this API is at resource instance level (GET /providers/Microsoft.Logic/Ise/ise1/skus. This again violates the ARM contract as the API should be GET /providers/Microsoft.Logic/skus. But seems like Azure monitor team(auto scale) has been asking teams to do this pattern. It will block teams onboarding to auto scale unless Autoscale engine changes to incorporate this. This needs to be worked out. For now approving this.

visharm commented:
> conclusion

ARM team has signed off, Could you please take a look?

yungezz commented:
will merge after CI passed.

yungezz commented:
Pls fix the comments which is from CI failure

visharm committed:
Added new version reference in readme files

visharm committed:
Fixed build errors

visharm committed:
Fixed build errors

visharm committed:
Fixed more errors

visharm committed:
fixed errors

visharm committed:
Fixed more errors

visharm committed:
Fixed more validation errors

visharm committed:
Fixed errors

visharm committed:
Fixed valiation errors in old files

visharm committed:
Fixed error

visharm committed:
Fixed error

visharm committed:
Fixed more errors

visharm committed:
more changes

visharm committed:
Fixed ISE errors

visharm committed:
fixed more errors

visharm committed:
Fixed ISE enum case

visharm committed:
added missing decriptions

visharm committed:
Fixed lintdiff warnings

visharm committed:
Fixed more warnings

visharm committed:
More fixes.

michaeljqzq commented on specification/logic/resource-manager/readme.md:
````diff
@@ -39,6 +39,16 @@ directive:
+These settings apply only when `--package-2019-05` is specified on the command line.
+
+``` yaml $(tag) == 'package-2019-05'
+-azure-validator: true
````
grammar mistake. You can refer to other RP as reference

michaeljqzq replied:
BTW, you can run linter on your local env by following command `npx autorest --validation --azure-validator --message-format=json specification/logic/resource-manager/readme.md`

visharm committed:
Removed extra characters

visharm committed:
Fixed typo

yungezz commented on specification/logic/resource-manager/readme.md:
```diff
@@ -39,6 +39,15 @@ directive:
 
+### Tag: package-2019-05
+
+These settings apply only when `--package-2019-05` is specified on the command line.
```
--tag=package-2019-05

visharm replied:
Fixed but still see the readme errors. What is the issue now?

visharm committed:
Fixed readme

visharm committed:
fixed readme

visharm committed:
fixed readme

visharm committed:
reverted change

michaeljqzq commented on specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:
```diff
@@ -0,0 +1,12665 @@
+          }
+        ],
+        "responses": {
+          "202": {
```
There is no "200" response here, which cause the lintdiff errror

visharm replied:
This API does not return 200. Its on old API why would this cause lintdiff error, it should cause model validation error.

jhendrixMSFT replied:
Unclear how API version from 2019 is an old API.  At any rate if the behavior cannot be changed the linter error can be suppressed with approval from ARM.

visharm replied:
These are the old APIs just added to 2019 version.

ArcturusZhang commented on specification/logic/resource-manager/readme.go.md:
````diff
@@ -13,12 +13,22 @@ go:
 
 ``` yaml $(go) && $(multiapi)
 batch:
+  - tag: package-2019-05-01
````
Please change this tag to `package-2019-05` as it is in readme.md

ArcturusZhang commented on specification/logic/resource-manager/readme.go.md:
````diff
@@ -13,12 +13,22 @@ go:
+These settings apply only when `--tag=package-2019-05-01 --go` is specified on the command line.
+Please also specify `--go-sdk-folder=<path to the root directory of your azure-sdk-for-go clone>`.
+
+``` yaml $(tag) == 'package-2019-05-01' && $(go)
````
Please change this tag to `package-2019-05` as it is in readme.md

ArcturusZhang reviewed (changes requested):
Please fix issues in readme.go.md file

visharm committed:
Fixed tags in readmes

ArcturusZhang commented:
> Please fix issues in readme.go.md file

Hi @jhendrixMSFT it seems that there is an issue related with code gen of go, I could not see where the problem locates, could you please help about this?

weidongxu-microsoft commented on specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:
```diff
@@ -0,0 +1,12665 @@
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
```
Please check whether it is intended. It specifies "nextLink" while there seems no response body defined (apart from the 202).

jhendrixMSFT replied:
This is likely why Go code generation is crashing.  If this operation doesn't return a model please remove this element.

visharm committed:
Fixed error

visharm committed:
Fixed model errors

visharm committed:
Fixed more errors in Logic json

jhendrixMSFT commented on specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json:
```diff
@@ -0,0 +1,12662 @@
+        "tags": [
+          "WorkflowRunActions"
+        ],
+        "operationId": "WorkflowRunActionRepetitio
```
The new linter error for this is due to the properties missing a nested `properties`, take a look at the definition for `Workflow` above.

visharm committed:
Fixed resource reference issue

visharm committed:
Fixed Resubmit trigger histories
